platform/mbed_mktime.h

```diff
@@ -33,22 +33,42 @@ extern "C" {
  * @{
```

Comment:
All these functions are with a `_` prefix and marked for hal usage. Should we point users to how apps can use this functions or at least get this functionality?

Reply:
Do you suggest to reduce description for these functions?

Reply:
No, I'm just saying that as an user I would go around the sources, I would find these functions and read that it should only be used in hal without pointing me to alternative. Also we add them to user documentation via doxygen.

Reply:
As far as I understand the purpose of these functions is for RTC drivers provided by vendors, so maybe we should remove them from user documentation.

Reply:
Ok, it's there already and we should solve that problem across all the functions - aka separate user APIs and the internal stuff. Right now platform contains mix of both. Let's leave it as it is I'll create an action to have a look into that.